### PR TITLE
Include text warning of 0RTT properties

### PR DESCRIPTION
**Pull request by wbl**
Based on list conversations, I'm including some API guidance.

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -1056,6 +1056,12 @@ will not process the same data twice for the same connection) and
 an attacker will not be able to make 0-RTT data appear to be
 1-RTT data (because it is protected with different keys.)
 
+These properties mean that 0-RTT data must be specially treated by applications. They SHOULD be required
+to opt-in to 0-RTT data transmission or recieving it: implementations SHOULD NOT preappend recieved 0-RTT data
+to 1-RTT data automatically,but provide special functions to examine 0-RTT data.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: missing space after comma

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -1056,6 +1056,12 @@ will not process the same data twice for the same connection) and
 an attacker will not be able to make 0-RTT data appear to be
 1-RTT data (because it is protected with different keys.)
 
+These properties mean that 0-RTT data must be specially treated by applications. They SHOULD be required
+to opt-in to 0-RTT data transmission or recieving it: implementations SHOULD NOT preappend recieved 0-RTT data
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: 'I' before 'E' except after 'C', so the correct spellings are "receiving" and "received".
"preappend" should be "prepend"

(also, wrap lines to around 80 characters, please)

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -1056,6 +1056,12 @@ will not process the same data twice for the same connection) and
...
+to 1-RTT data automatically,but provide special functions to examine 0-RTT data.
+
+Protocols on top of TLS 1.3 should carefully consider when 0-RTT can be used.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: This sentence is a bit superfluous down here. Might be better to put a comma at the end of the first new sentence up top and stick this sentence fragment there. e.g. "... treated by applications, and protocols on top of ..."

**Comment by martinthomson** [comment]
I like that you have included a requirement to have a profile for each protocol that intends to use 0-RTT, that's a fine thing to have (and it will make some people happier), but the first paragraph is a bit woolly.  You have a lowercase must, followed by a few SHOULD/SHOULD NOT statements.  You really need to be clearer and more definitive here:

> Protocols MUST NOT use 0-RTT data without a profile that defines its use.  That profile needs to identify which messages or interactions are safe to use with 0-RTT.
>
> In addition, to avoid accidental misuse, implementations SHOULD NOT enable 0-RTT unless specifically requested.  Special functions for 0-RTT data are RECOMMENDED to ensure that an application is always aware that it is sending or receiving data that might be replayed.

**Comment by ekr** [comment]
I adopted Martin Thomson's text.

**Pull request closed without merging**